src/main/java/net/neoforged/neoforge/network/PacketDistributor.java

```diff
@@ -8,256 +8,133 @@
-     * 
-     * @param input The input to apply
-     * @return A curried instance
+     * Send the given payload(s) to all players in the given dimension
      */
-    public PacketTarget with(T input) {
-        return new PacketTarget(functor.apply(this, input), this);
+    public static void sendToPlayersInDimension(ResourceKey<Level> dimension, CustomPacketPayload... payloads) {
+        getServer().getPlayerList().broadcastAll(makeClientboundPacket(payloads), dimension);
     }
 
     /**
-     * Apply a no argument value to a distributor to generate an instance for sending packets to.
-     *
-     * @see #ALL
-     * @see #SERVER
-     * @return A curried instance
+     * Send the given payload(s) to all players near the given {@link TargetPoint}
      */
-    public PacketTarget noArg() {
-        return new PacketTarget(functor.apply(this, null), this);
-    }
-
-    private Consumer<Packet<?>> playerConsumer(final ServerPlayer entityPlayerMP) {
-        return p -> entityPlayerMP.connection.send(p);
+    public static void sendToPlayersNear(TargetPoint target, CustomPacketPayload... payloads) {
+        Packet<?> packet = makeClientboundPacket(payloads);
+        getServer().getPlayerList().broadcast(target.excluded, target.x, target.y, target.z, target.r, target.dim, packet);
     }
 
-    private Consumer<Packet<?>> playerListDimConsumer(final ResourceKey<Level> dimensionType) {
-        return p -> getServer().getPlayerList().broadcastAll(p, dimensionType);
-    }
-
-    private Consumer<Packet<?>> playerListAll() {
-        return p -> getServer().getPlayerList().broadcastAll(p);
+    /**
+     * Send the given payload(s) to all players on the server
+     */
+    public static void sendToAllPlayers(CustomPacketPayload... payloads) {
+        getServer().getPlayerList().broadcastAll(makeClientboundPacket(payloads));
     }
 
-    private Consumer<Packet<?>> clientToServer() {
-        return p -> Objects.requireNonNull(Minecraft.getInstance().getConnection()).send(p);
+    /**
+     * Send the given payload(s) to all players tracking the given entity
+     */
+    public static void sendToPlayersTrackingEntity(Entity entity, CustomPacketPayload... payloads) {
+        ((ServerChunkCache) entity.level().getChunkSource()).broadcast(entity, makeClientboundPacket(payloads));
     }
 
-    private Consumer<Packet<?>> playerListPointConsumer(final TargetPoint targetPoint) {
-        return p -> {
-            final TargetPoint tp = targetPoint;
-            getServer().getPlayerList().broadcast(tp.excluded, tp.x, tp.y, tp.z, tp.r2, tp.dim, p);
-        };
+    /**
+     * Send the given payload(s) to all players tracking the given entity and the entity itself if it is a player
+     */
+    public static void sendToPlayersTrackingEntityAndSelf(Entity entity, CustomPacketPayload... payloads) {
+        ((ServerChunkCache) entity.level().getChunkSource()).broadcastAndSend(entity, makeClientboundPacket(payloads));
```

Comment:
Since this is just `Entity`, we should do an instanceof and throw a proper exception if called on the wrong side instead of directly casting the chunk cache